modin/engines/dask/pandas_on_dask/frame/partition.py

```diff
@@ -80,25 +71,42 @@ def apply(self, func, **kwargs):
              A new `BaseFramePartition` containing the object that has had `func`
              applied to it.
         """
-        func = pkl.dumps(func)
-        call_queue = self.call_queue + [[func, kwargs]]
-        future = get_client().submit(
-            apply_list_of_funcs, call_queue, self.future, pure=False
-        )
-        futures = [get_client().submit(lambda l: l[i], future) for i in range(2)]
+        client = get_client()
+        call_queue = self.call_queue + [[func, args, kwargs]]
+        if len(call_queue) > 1:
+            future = client.submit(
+                apply_list_of_funcs, self.future, call_queue, pure=False
+            )
+        else:
+            func, args, kwargs = call_queue[0]
+            future = client.submit(apply_func, self.future, func, *args, **kwargs)
+        futures = [
+            client.submit(lambda l, i: l[i], future, i, pure=False) for i in range(2)
+        ]
         return PandasOnDaskFramePartition(futures[0], ip=futures[1])
 
-    def add_to_apply_calls(self, func, **kwargs):
+    def add_to_apply_calls(self, func, *args, **kwargs):
```

Comment:
I see a `*args` change here also, are we using it? In the past, I have only used `**kwargs` to aid in debugging.

Reply:
See the comment ([link](https://github.com/modin-project/modin/pull/2919#discussion_r639504163)).